hoodie-common/src/main/java/com/uber/hoodie/common/model/FileSlice.java

```diff
@@ -44,17 +47,32 @@
    */
   private HoodieDataFile dataFile;
 
+  /**
+   * Outstanding Compaction request instant time
+   * For a given file-slice, only one compaction request can be outstanding.
+   */
+  private String outstandingCompactionInstant;
```

Comment:
I am wondering if this kind of information should be leaked into the FileSlice level? I think @n3nash mentioned something similar. 
I feel we can leave the FileSlice with an outstandingCompaction with the baseCommitTime = outstandingCompactionTme and just figure out the log files for append at the higher FileSystemView level?

Reply:
@vinothchandar / @n3nash :  
It makes sense to not track outstandingCompactionCommit in File-Slice.  Just so that we are all on same page : I am also going to make the change to remove the FileGroupBuilder. File-Slices and File-Groups will be untouched. File Slices are un-merged when constructing file-groups. A file slice will be defined as a collection of base-file (optional) and one or more log-files whose compaction would result in one parquet-file.  At the FileSystemView level where we return latest-file slices for realtime-readers (For eg: RealtimeView.getLatestFileSlices()), we need to merge the file-slice after pending compaction-request with the one before by tracking all the file-Ids for which compaction is scheduled. Nishith's concerns should be addressed with this change.

Reply:
sounds good to me.